freeipa/src/main/java/com/sequenceiq/freeipa/flow/freeipa/downscale/action/FreeIpaDownscaleActions.java

```diff
@@ -432,11 +431,6 @@ protected StackContext createFlowContext(FlowParameters flowParameters, StateCon
                 return super.createFlowContext(flowParameters, stateContext, payload);
             }
 
-            @Override
```

Comment:
let me try this out. But I would say ideally you should have no such logic in the failure handling part which can fail constantly in my opinion. This would indicate that the failure handling action was never tested

Reply:
I have tested with 2 different flows and there is no loop. It would fail with the following, and the flow would stop:
```
021-09-17 12:01:33,985 [reactorDispatcher-10] lambda$reactor$1:145 ERROR c.s.f.r.c.EventBusConfig - [type:STACK] [crn:crn:cdp:freeipa:us-west-1:cloudera:freeipa:e28e9559-e222-40de-b033-ebdacb44e866] [name:mmolnar-awsenv2-freeipa] [flow:1f1a91e2-7c8f-46bd-ad1b-2b79983549ca] [requestid:c480ed03-a041-4d71-bacf-a606dadf113c] [tenant:cloudera] [userCrn:crn:altus:iam:us-west-1:cloudera:user:mmolnar@cloudera.com] [environment:crn:cdp:environments:us-west-1:cloudera:environment:6ce22431-6f79-4b4d-981f-07ba899db7ad] [traceId:0465a361c0ec7006] [spanId:38ce7e0130802f45] [operationId:] Exception happened in dispatcher
com.sequenceiq.cloudbreak.common.exception.CloudbreakServiceException: Missing error handling for com.sequenceiq.freeipa.flow.freeipa.salt.update.action.SaltUpdateFailureAction
	at com.sequenceiq.flow.core.AbstractAction.lambda$execute$0(AbstractAction.java:111)
	at com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider.doAs(ThreadBasedUserCrnProvider.java:113)
	at com.sequenceiq.flow.core.AbstractAction.execute(AbstractAction.java:78)
```
If there is no failure payload for that action it would cause a nullpointer, so I think this is way better

Reply:
Thanks for tying. Now I see why: the failure handler has no failureEvent. I overlooked it previously.

https://github.com/hortonworks/cloudbreak/blob/16067565792b2db2d6f80e7098753b7ed3625523/flow/src/main/java/com/sequenceiq/flow/core/AbstractAction.java#L107